rust/integration-tests/src/lib.rs

```diff
@@ -0,0 +1,121 @@
+use anyhow::Context;
+use aptos_protos::transaction::v1::Transaction;
+use diesel::{pg::PgConnection, sql_query, Connection, RunQueryDsl};
+use itertools::Itertools;
+use processor::{
+    processors::{ProcessorConfig, ProcessorTrait},
+    utils::database::{new_db_pool, run_pending_migrations},
+    worker::build_processor_for_testing,
+};
+use testcontainers::{
+    core::{IntoContainerPort, WaitFor},
+    runners::AsyncRunner,
+    ContainerAsync, GenericImage, ImageExt,
+};
+
+mod simple_tests;
+
+/// The test context struct holds the test name and the transaction batches.
+pub struct TestContext {
+    pub transaction_batches: Vec<Transaction>,
+    postgres_container: ContainerAsync<GenericImage>,
+}
+
+#[derive(Debug, Clone)]
+pub struct TestProcessorConfig {
+    config: ProcessorConfig,
+}
+
+impl TestContext {
+    // TODO: move this to builder pattern to allow chaining.
+    pub async fn new(txn_bytes: &[&[u8]]) -> anyhow::Result<Self> {
+        let transaction_batches = txn_bytes
+            .iter()
+            .enumerate()
+            .map(|(idx, txn)| {
+                let mut txn: Transaction = serde_json::from_slice(txn).unwrap();
+                txn.version = idx as u64;
+                txn
+            })
+            .collect::<Vec<Transaction>>();
+        let postgres_container = GenericImage::new("postgres", "14")
+            .with_exposed_port(5432.tcp())
+            .with_wait_for(WaitFor::message_on_stderr(
+                "database system is ready to accept connections",
+            ))
+            .with_env_var("POSTGRES_DB", "postgres")
+            .with_env_var("POSTGRES_USER", "postgres")
+            .with_env_var("POSTGRES_PASSWORD", "postgres")
+            .start()
+            .await
+            .expect("Redis started");
```

Comment:
nit: Postgres started

Reply:
Testing is failing here, is there pre-requisites I have to run before run `cargo test`?
![Screenshot 2024-08-02 at 12 24 21 PM](https://github.com/user-attachments/assets/29314204-7dad-44cb-a593-939161ef096a)